### PR TITLE
Refactor/options logger

### PR DESCRIPTION
**Pull request by sudden6**
- [x] Commits follow our [git commit guidelines](https://github.com/qTox/qTox/blob/master/CONTRIBUTING.md#git-commit-guidelines)

@Diadlo Can you review if this is a good idea?

---
This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/5193)

### CODE REVIEW

File: src/core/toxoptions.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        qWarning() << "Core starting with IPv6 disabled. LAN discovery may not work properly.";
+    }
+
+    Tox_Options* tox_opts = tox_options_new(nullptr);
+
+    if(!tox_opts) {
+        return {};
+    }
+
+    auto toxOptions = std::unique_ptr<ToxOptions>(new ToxOptions(tox_opts, proxyAddr.toUtf8()));
```
Comment (Diadlo) [inline src/core/toxoptions.cpp:74]: Why do you use pointer here? We can create `ToxOptions` on stack and use `std::move` at the end of the function. What do you think?

**Review by Diadlo — COMMENTED** [review]
Mostly logic of change LGTM. Only one question (see below)

File: src/core/toxoptions.cpp (hunk 1 above)
Comment (sudden6) [reply src/core/toxoptions.cpp:74]: Because the allocation of the internal Tox_Options struct can fail and `nullptr` is used to represent that.

**Comment by Diadlo** [comment]
:lgtm_strong:

---

Reviewed 6 of 6 files at r1.
Review status: :shipit: complete! 1 of 1 LGTMs obtained

---



*Comments from [Reviewable](https://reviewable.io/reviews/qtox/qtox/5193#-:-LFwL9jZBYyYj7JhWM9M:b-jbsf8z)*

**Pull request merged**